agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -270,3 +275,45 @@ func networkModeTest(t *testing.T, agent *TestAgent, mode string) error {
 	}
 	return nil
 }
+
+// TestCustomAttributesWithMaxOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with maximum number of supported key, value pairs
+func TestCustomAttributesWithMaxOptions(t *testing.T) {
+	maxAttributes := 10
+	customAttributes := `{
+                "key1": "val1",
+                "key2": "val2",
+                "key3": "val3",
+                "key4": "val4",
+                "key5": "val5",
+                "key6": "val6",
+                "key7": "val7",
+                "key8": "val8",
+                "key9": "val9",
+                "key0": "val0"
+        }`
+	os.Setenv("ECS_INSTANCE_ATTRIBUTES", customAttributes)
+	defer os.Unsetenv("ECS_INSTANCE_ATTRIBUTES")
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	params := &ecsapi.DescribeContainerInstancesInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+	}
+
+	resp, err := ECS.DescribeContainerInstances(params)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.ContainerInstances)
+	require.Len(t, resp.ContainerInstances, 1)
+
+	attribMap := AttributesToMap(resp.ContainerInstances[0].Attributes)
+	assert.NotEmpty(t, attribMap)
+
+	for i := 0; i < maxAttributes; i++ {
```

Comment:
I think it would be much better, you should check the variable you defined, but not the content.
```
for k, v := range customAttributes{
assert.Equal(t, v, attribMap[k], "Values should match")
}
```